src/defaults/runningTerminals.tsx

```diff
@@ -0,0 +1,112 @@
+import * as ReactDOM from 'react-dom';
+import * as React from 'react';
+
+import {
+    JupyterLabPlugin,
+    JupyterLab,
+    ApplicationShell
+} from '@jupyterlab/application';
+import { IDefaultStatusesManager } from './manager';
+
+import { TerminalManager, TerminalSession } from '@jupyterlab/services';
+
+import { Widget } from '@phosphor/widgets';
+
+export namespace RunningTerminalComponent {
+    export interface IState {
+        terminalSession: number;
+    }
+    export interface IProps {
+        terminalManager: TerminalManager;
+        shellHost: ApplicationShell;
+    }
+}
+
+export class RunningTerminalComponent extends React.Component<
+    RunningTerminalComponent.IProps,
+    RunningTerminalComponent.IState
+> {
+    state = {
+        terminalSession: 0
+    };
+
+    constructor(props: RunningTerminalComponent.IProps) {
+        super(props);
+        this.props.terminalManager.runningChanged.connect(this.updateTerminal);
+    }
+
+    componentDidMount() {
+        TerminalSession.listRunning().then(value =>
```

Comment:
Can you use the `this.props.terminalManager` for the call to `listRunning` instead of the global object

Reply:
no